feat: sync `CommunitySettings` across paired devices

0x-r4bbit opened the pull request:
This commit introduces a new `synced_at` field in the
`communities_settings` table so that it can be leveraged for syncing
community settings across devices.

It also adds sending and handling of sync community settings where
appropriate.

0x-r4bbit committed:
feat: sync `CommunitySettings` across paired devices

This commit introduces a new `clock` field in the
`communities_settings` table so that it can be leveraged for syncing
community settings across devices.

It alsoe exends existing `syncCommunity` APIs to generate
`SyncCommunitySettings` as well, avoiding sending additional sync messages
for community settings.

When editing communities however, we still sync community settings
explicitly are we aren't syncing the community itself in that case.

ghost commented:
<!--prchecklist--> 
### Pull Request Checklist
- [ ] Have you updated the documentation, if impacted (e.g. [docs.status.im](https://docs.status.im/docs/build_status_go.html))?
- [ ] Have you tested changes with mobile?
- [ ] Have you tested changes with desktop?

0x-r4bbit commented:
@richard-ramos @Samyoul only the first commit needs review, the second one is build artifacts from `make generate`. I'll squash them into the first one, once reviewed.

status-im-auto commented:
### Jenkins Builds
<details>
<summary>Click to see older builds (27)</summary>

| :grey_question: | Commit | :hash: | Finished (UTC) | Duration | Platform | Result |
|-|-|-|-|-|-|-|
| :x: | 7c8a6271 | [#1](https://ci.status.im/job/status-go/job/prs/job/linux/job/PR-2701/1/) | 2022-06-01 08:02:18 | ~1 min | `linux` | [:page_facing_up:`log`](https://ci.status.im/job/status-go/job/prs/job/linux/job/PR-2701/1/consoleText) |
| :x: | 7c8a6271 | [#1](https://ci.status.im/job/status-go/job/prs/job/android/job/PR-2701/1/) | 2022-06-01 08:03:00 | ~2 min | `android` | [:page_facing_up:`log`](https://ci.status.im/job/status-go/job/prs/job/android/job/PR-2701/1/consoleText) |
| :x: | 7c8a6271 | [#1](https://ci.status.im/job/status-go/job/prs/job/ios/job/PR-2701/1/) | 2022-06-01 08:03:18 | ~2 min | `ios` | [:page_facing_up:`log`](https://ci.status.im/job/status-go/job/prs/job/ios/job/PR-2701/1/consoleText) |
| | | | | | | |
| :x: | 15e2e53c | [#2](https://ci.status.im/job/status-go/job/prs/job/ios/job/PR-2701/2/) | 2022-06-01 08:03:50 | ~30 sec | `ios` | [:page_facing_up:`log`](https://ci.status.im/job/status-go/job/prs/job/ios/job/PR-2701/2/consoleText) |
| :x: | 15e2e53c | [#2](https://ci.status.im/job/status-go/job/prs/job/linux/job/PR-2701/2/) | 2022-06-01 08:04:12 | ~1 min | `linux` | [:page_facing_up:`log`](https://ci.status.im/job/status-go/job/prs/job/linux/job/PR-2701/2/consoleText) |
| :x: | 15e2e53c | [#2](https://ci.status.im/job/status-go/job/prs/job/android/job/PR-2701/2/) | 2022-06-01 08:05:32 | ~2 min | `android` | [:page_facing_up:`log`](https://ci.status.im/job/status-go/job/prs/job/android/job/PR-2701/2/consoleText) |
| | | | | | | |
| :x: | 7e422b7e | [#3](https://ci.status.im/job/status-go/job/prs/job/ios/job/PR-2701/3/) | 2022-06-01 11:18:05 | ~48 sec | `ios` | [:page_facing_up:`log`](https://ci.status.im/job/status-go/job/prs/job/ios/job/PR-2701/3/consoleText) |
| :x: | 7e422b7e | [#3](https://ci.status.im/job/status-go/job/prs/job/android/job/PR-2701/3/) | 2022-06-01 11:18:25 | ~1 min | `android` | [:page_facing_up:`log`](https://ci.status.im/job/status-go/job/prs/job/android/job/PR-2701/3/consoleText) |
| :x: | 7e422b7e | [#3](https://ci.status.im/job/status-go/job/prs/job/linux/job/PR-2701/3/) | 2022-06-01 11:18:34 | ~1 min | `linux` | [:page_facing_up:`log`](https://ci.status.im/job/status-go/job/prs/job/linux/job/PR-2701/3/consoleText) |
| | | | | | | |
| :heavy_check_mark: | 0138018d | [#4](https://ci.status.im/job/status-go/job/prs/job/linux/job/PR-2701/4/) | 2022-06-01 12:32:33 | ~1 min | `linux` | [:package:`zip`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-desktop-220601-123105-013801-pr2701.zip) |
| :heavy_check_mark: | 0138018d | [#4](https://ci.status.im/job/status-go/job/prs/job/ios/job/PR-2701/4/) | 2022-06-01 12:34:38 | ~3 min | `ios` | [:package:`zip`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-ios-220601-123105-013801-pr2701.zip) |
| :heavy_check_mark: | 0138018d | [#4](https://ci.status.im/job/status-go/job/prs/job/android/job/PR-2701/4/) | 2022-06-01 12:34:58 | ~3 min | `android` | [:package:`aar`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-android-220601-123105-013801-pr2701.aar) |
| | | | | | | |
| :heavy_check_mark: | 605598cd | [#5](https://ci.status.im/job/status-go/job/prs/job/linux/job/PR-2701/5/) | 2022-06-01 14:31:05 | ~1 min | `linux` | [:package:`zip`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-desktop-220601-142922-605598-pr2701.zip) |
| :heavy_check_mark: | 605598cd | [#5](https://ci.status.im/job/status-go/job/prs/job/ios/job/PR-2701/5/) | 2022-06-01 14:31:41 | ~2 min | `ios` | [:package:`zip`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-ios-220601-142922-605598-pr2701.zip) |
| :heavy_check_mark: | 605598cd | [#5](https://ci.status.im/job/status-go/job/prs/job/android/job/PR-2701/5/) | 2022-06-01 14:33:55 | ~4 min | `android` | [:package:`aar`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-android-220601-142922-605598-pr2701.aar) |
| | | | | | | |
| :heavy_check_mark: | 4a10b896 | [#6](https://ci.status.im/job/status-go/job/prs/job/linux/job/PR-2701/6/) | 2022-06-01 14:46:08 | ~1 min | `linux` | [:package:`zip`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-desktop-220601-144432-4a10b8-pr2701.zip) |
| :heavy_check_mark: | 4a10b896 | [#6](https://ci.status.im/job/status-go/job/prs/job/android/job/PR-2701/6/) | 2022-06-01 14:48:15 | ~3 min | `android` | [:package:`aar`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-android-220601-144432-4a10b8-pr2701.aar) |
| :heavy_check_mark: | 4a10b896 | [#6](https://ci.status.im/job/status-go/job/prs/job/ios/job/PR-2701/6/) | 2022-06-01 14:50:39 | ~6 min | `ios` | [:package:`zip`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-ios-220601-144432-4a10b8-pr2701.zip) |
| | | | | | | |
| :heavy_check_mark: | ba3aaec1 | [#7](https://ci.status.im/job/status-go/job/prs/job/linux/job/PR-2701/7/) | 2022-06-03 08:50:59 | ~7 min | `linux` | [:package:`zip`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-desktop-220603-084259-ba3aae-pr2701.zip) |
| :heavy_check_mark: | ba3aaec1 | [#7](https://ci.status.im/job/status-go/job/prs/job/android/job/PR-2701/7/) | 2022-06-03 08:54:46 | ~11 min | `android` | [:package:`aar`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-android-220603-084259-ba3aae-pr2701.aar) |
| :heavy_check_mark: | ba3aaec1 | [#7](https://ci.status.im/job/status-go/job/prs/job/ios/job/PR-2701/7/) | 2022-06-03 09:01:41 | ~18 min | `ios` | [:package:`zip`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-ios-220603-084259-ba3aae-pr2701.zip) |
| | | | | | | |
| :heavy_check_mark: | 0a0d7d44 | [#8](https://ci.status.im/job/status-go/job/prs/job/ios/job/PR-2701/8/) | 2022-06-03 09:07:23 | ~1 min | `ios` | [:package:`zip`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-ios-220603-090541-0a0d7d-pr2701.zip) |
| :heavy_check_mark: | 0a0d7d44 | [#8](https://ci.status.im/job/status-go/job/prs/job/linux/job/PR-2701/8/) | 2022-06-03 09:07:53 | ~2 min | `linux` | [:package:`zip`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-desktop-220603-090543-0a0d7d-pr2701.zip) |
| :heavy_check_mark: | 0a0d7d44 | [#8](https://ci.status.im/job/status-go/job/prs/job/android/job/PR-2701/8/) | 2022-06-03 09:10:00 | ~4 min | `android` | [:package:`aar`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-android-220603-090540-0a0d7d-pr2701.aar) |
| | | | | | | |
| :heavy_check_mark: | 4fe3daee | [#9](https://ci.status.im/job/status-go/job/prs/job/linux/job/PR-2701/9/) | 2022-06-07 15:20:22 | ~6 min | `linux` | [:package:`zip`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-desktop-220607-151341-4fe3da-pr2701.zip) |
| :heavy_check_mark: | 4fe3daee | [#9](https://ci.status.im/job/status-go/job/prs/job/ios/job/PR-2701/9/) | 2022-06-07 15:21:13 | ~7 min | `ios` | [:package:`zip`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-ios-220607-151341-4fe3da-pr2701.zip) |
| :heavy_check_mark: | 4fe3daee | [#9](https://ci.status.im/job/status-go/job/prs/job/android/job/PR-2701/9/) | 2022-06-07 15:22:46 | ~9 min | `android` | [:package:`aar`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-android-220607-151338-4fe3da-pr2701.aar) |
</details>

| :grey_question: | Commit | :hash: | Finished (UTC) | Duration | Platform | Result |
|-|-|-|-|-|-|-|
| :heavy_check_mark: | a5c81e94 | [#10](https://ci.status.im/job/status-go/job/prs/job/linux/job/PR-2701/10/) | 2022-06-09 10:31:04 | ~1 min | `linux` | [:package:`zip`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-desktop-220609-102913-a5c81e-pr2701.zip) |
| :heavy_check_mark: | a5c81e94 | [#10](https://ci.status.im/job/status-go/job/prs/job/ios/job/PR-2701/10/) | 2022-06-09 10:32:57 | ~3 min | `ios` | [:package:`zip`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-ios-220609-102911-a5c81e-pr2701.zip) |
| :heavy_check_mark: | a5c81e94 | [#10](https://ci.status.im/job/status-go/job/prs/job/android/job/PR-2701/10/) | 2022-06-09 10:34:27 | ~5 min | `android` | [:package:`aar`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-android-220609-102911-a5c81e-pr2701.aar) |
| | | | | | | |
| :heavy_check_mark: | 0160fc97 | [#11](https://ci.status.im/job/status-go/job/prs/job/ios/job/PR-2701/11/) | 2022-06-16 10:33:16 | ~3 min | `ios` | [:package:`zip`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-ios-220616-102954-0160fc-pr2701.zip) |
| :heavy_check_mark: | 0160fc97 | [#11](https://ci.status.im/job/status-go/job/prs/job/linux/job/PR-2701/11/) | 2022-06-16 10:33:38 | ~3 min | `linux` | [:package:`zip`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-desktop-220616-102954-0160fc-pr2701.zip) |
| :heavy_check_mark: | 0160fc97 | [#11](https://ci.status.im/job/status-go/job/prs/job/android/job/PR-2701/11/) | 2022-06-16 10:34:12 | ~4 min | `android` | [:package:`aar`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-android-220616-102953-0160fc-pr2701.aar) |

0x-r4bbit commented on protocol/protobuf/application_metadata_message.proto:
```diff
@@ -61,5 +61,6 @@ message ApplicationMetadataMessage {
     SYNC_WALLET_ACCOUNT = 45;
     ACCEPT_CONTACT_REQUEST = 46;
     RETRACT_CONTACT_REQUEST = 47;
+    SYNC_COMMUNITY_SETTINGS = 49;
```
There's another PR pending that introduces the `48` field here, so I already went with `49`

0x-r4bbit commented:
@cammellos I've updated this PR according to our discussion offline, namely:

1. Changed `synced_at` field to `clock` field in `CommunitySettings`
2. Extended `SynCommunity` to have a `settings` field which is of type `SyncCommunitySettings`, allowing us to send the settings along with a `SyncCommunity` message which reduces bandwidth
3. Notice that we still have an explicit sync message for communities settings which is used when editing a community, because in that case we're not sending sync messages for the community itself
3. Updated the tests accordingly, also added an explicit test for syncing community settings when editing communities as it needs special treatment

Still split up the changes in two commits, so reviewing the first one should be enough.

Samyoul reviewed (approved):
Nice PR, good tests

0x-r4bbit commented:
Looking into the failing test

0x-r4bbit commented:
Okay so it seems the issue is that there's no community in alice second device that has a synced_at value > 0, which means, alice second device might get the community description from the chat message (in which bob shares the community), but it has not received any trigger to get synced (which is required for it to have a sync_at >0)

Investigating further..

0x-r4bbit commented:
Alright, turned out that handling sync community settings was the culprit because we were sending malformed data.
Pushing an update

0x-r4bbit commented:
@churik Not sure if required, but pinging to this can be tested as per https://notes.status.im/pipeline-status-go

qoqobolo commented:
Hi @PascalPrecht, e2e tests have caught the crash on the mobile app in this PR.
Take a look, please. 

### ISSUE 1: The app crashes when recovering multiaccount with a seed phrase

**Steps to reproduce:**
1. Install PR build from https://github.com/status-im/status-react/pull/13448#issuecomment-1145951415
2. Recover multiaccount with a seed phrase
3. Tap `Let's go`

**Logs:**
[crashlog_recoverseed.log](https://github.com/status-im/status-go/files/8833129/crashlog_recoverseed.log)


https://user-images.githubusercontent.com/67952253/171877366-870981b3-fb8e-4060-b0fd-137ddc22ef94.MP4

0x-r4bbit commented:
Hey @qoqobolo thanks for letting me know!
I'll have a look at it today

0x-r4bbit commented:
@qoqobolo Hey :)

I could reproduce the bug and updated the PR accordingly.
Would you mind running your battery of tests again?

qoqobolo commented:
> Would you mind running your battery of tests again?

Sure! I'll get back with the results.

qoqobolo commented:
@PascalPrecht thanks for the fix! Works perfectly now.

0x-r4bbit commented:
@qoqobolo Thanks for confirming!

@Samyoul does this need a bump in `VERSION` before merge?
